File: backend/src/main/java/com/ecommerce/backend/service/TenantService.java
package com.ecommerce.backend.service;

import com.ecommerce.backend.dto.request.CreateTenantRequest;
import com.ecommerce.backend.dto.response.TenantResponse;
import org.springframework.data.domain.Page;

import java.util.List;

public interface TenantService {

    TenantResponse createTenant(CreateTenantRequest request);

    List<TenantResponse> getAllTenants();

    Page<TenantResponse> getAllTenantsPage(int page, int size, String keyword);

    TenantResponse getTenantById(Long tenantId);

    TenantResponse updateTenant(Long tenantId, CreateTenantRequest request);

    void deleteTenant(Long tenantId);

    void assignTenantAdmin(Long tenantId, Long userId);
}

File: frontend/src/services/tenantService.js
import api from "./api";

export const getTenants = async () => {
  return await api.get(`/tenants`);
};

export const getTenantsPage = async (page = 0, size = 6, keyword = "") => {
  return await api.get(`/tenants/page`, {
    params: { page, size, keyword },
  });
};

export const createTenant = async (tenant) => {
  return await api.post(`/tenants`, tenant);
};

export const getTenantById = async (tenantId) => {
  return await api.get(`/tenants/${tenantId}`);
};

export const updateTenant = async (tenantId, tenant) => {
  return await api.put(`/tenants/${tenantId}`, tenant);
};

export const deleteTenant = async (tenantId) => {
  return await api.delete(`/tenants/${tenantId}`);
};

export const assignTenantAdmin = async (tenantId, userId) => {
  return await api.post(`/tenants/${tenantId}/assign-admin/${userId}`);
};